Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1426,6 +1427,23 @@ - (void)_runningOrTerminatingServiceWorkerCountForTesting:(void(^)(NSUInteger))c
     });
 }
 
+NSString * const WKLocalStorage = @"WKLocalStorage";
+
+- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSData *))completionHandler
+{
+    if ([dataTypes containsObject:WKLocalStorage]) {
+        _websiteDataStore->fetchLocalStorage([completionHandler = makeBlockPtr(completionHandler)](HashMap<WebCore::ClientOrigin, HashMap<String, String>> localStorage) {
+            WTF::Persistence::Encoder encoder;
+            encoder << localStorage;
```

Comment:
We should encode a version number first. Otherwise, we'll have trouble with backwards compatibility in the future.